xapian-core/api/valuerangeproc.cc

```diff
@@ -432,8 +432,120 @@ NumberRangeProcessor::operator()(const string& b, const string& e)
     }
 
     return RangeProcessor::operator()(
-	    b.empty() ? b : Xapian::sortable_serialise(num_b),
-	    e.empty() ? e : Xapian::sortable_serialise(num_e));
+	b.empty() ? b : Xapian::sortable_serialise(num_b),
+	e.empty() ? e : Xapian::sortable_serialise(num_e));
+
+not_our_range:
+    return Xapian::Query(Xapian::Query::OP_INVALID);
+}
+
+static const unsigned int size_map[26] = {
+    0,
+    1, // B
+    0, 0, 0, 0,
+    1024 * 1024 * 1024, // G
+    0, 0, 0,
+    1024, // K
+    0,
+    1024 * 1024, // M
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+static unsigned int
+get_size_multiplier_from_suffix(const char ch)
+{
+    // It takes character as input and return number of bytes
```

Comment:
The `It` sounds odd here.

Also, this barely goes beyond describing the function at a syntactic level.  One can see from the parameter list that it takes a character as input, and from the return type that it returns a number.  So all you've really told us is that the return value is measured in bytes (but actually it's not bytes but a scale factor - it could equally apply to bits as bytes).

What's more interesting to tell future developers (often including your future self) is the higher level semantics which aren't obvious from the function signature - what the function is for, e.g. the meaning of the output in relation to the input.  And handling of errors is important to document too.

It doesn't have to be an essay - indeed a good comments is often brief but to the point.  Something like this captures the important details: `Maps a file size suffix character to a scale factor.  Returns 0 for an unknown suffix character.`

If this was external API documentation, the list of recognised characters definitely ought to be in there too, but here that's already documented by the comments in `size_map[]` above - repeating that list here mostly means there are two versions which creates the risk of a future change updating one but not the other (https://en.wikipedia.org/wiki/Don't_repeat_yourself).  You could move the whole definition of `size_map[]` inside the function to make the connection clearer (and the code a little cleaner).